### PR TITLE
podman-desktop: init at 0.12.0

### PR DESCRIPTION
**Pull request by panda2134**
###### Description of changes

Added [podman-desktop](https://github.com/containers/podman-desktop), which is a GUI frontend for podman.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by teutat3s** [comment]
Result of `nixpkgs-review pr 222727` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>podman-desktop</li>
  </ul>
</details>

**Review by teutat3s — APPROVED** [review]

File: pkgs/applications/virtualization/podman-desktop/default.nix [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+, fixup_yarn_lock
+, electron_23
+, makeDesktopItem
+,
+}:
+
+stdenv.mkDerivation (self: {
+  pname = "podman-desktop";
+  version = "0.12.0";
+  name = "${self.pname}-${self.version}";
```
Comment (NickCao) [inline pkgs/applications/virtualization/podman-desktop/default.nix:18]: No need to set name explicitly.

File: pkgs/applications/virtualization/podman-desktop/default.nix [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+, fetchYarnDeps
+, desktopToDarwinBundle
+, fetchFromGitHub
+, fixup_yarn_lock
+, electron_23
+, makeDesktopItem
+,
+}:
+
+stdenv.mkDerivation (self: {
```
Comment (NickCao) [inline pkgs/applications/virtualization/podman-desktop/default.nix:15]: Doesn't mkYarnPackage work?

Comment (panda2134) [reply pkgs/applications/virtualization/podman-desktop/default.nix:15]: The code is generally migrated from the nix expression for `element-desktop`. I tried `mkYarnPackage` at first, but it seems to be triggering a bug in the dependency resolution of yarn. To be concise, the package `ws` resolved to a version that was not cached by `fetchYarnDeps`. Hence, I have to fix the lock file manually by applying a patch.

File: pkgs/applications/virtualization/podman-desktop/default.nix (hunk 1 above)
Comment (panda2134) [reply pkgs/applications/virtualization/podman-desktop/default.nix]: fixed.

File: pkgs/applications/virtualization/podman-desktop/default.nix [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  };
+
+  offlineCache = fetchYarnDeps {
+    yarnLock = "${self.src}/yarn.lock";
+    sha256 = "sha256-x0hqNxi6r1i3vBe1tJQl+Oht2St9VIH3Eq27MZLkojA=";
+  };
+
+  patches = [
+    ./patches/extension-no-download-podman.patch # podman should be installed with nix; disable auto-installation
+    ./patches/fix-yarn-lock-deterministic.patch
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/podman-desktop/default.nix:33]:
```suggestion
    # podman should be installed with nix; disable auto-installation
    ./extension-no-download-podman.patch 
    ./fix-yarn-lock-deterministic.patch
```

File: pkgs/applications/virtualization/podman-desktop/default.nix [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+      -c.electronVersion=${electron.version}
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p "$out/opt/podman-desktop"
+    cp -r dist/*-unpacked/{locales,resources{,.pak}} "$out/opt/podman-desktop"
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/podman-desktop/default.nix:85]:
```suggestion
    mkdir -p "$out/share/lib/podman-desktop"
    cp -r dist/*-unpacked/{locales,resources{,.pak}} "$out/share/lib/podman-desktop"
```
we usually use something like that

File: pkgs/applications/virtualization/podman-desktop/default.nix [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+  src = fetchFromGitHub {
+    owner = "containers";
+    repo = "podman-desktop";
+    rev = "v${self.version}";
+    sha256 = "sha256-gEjcI+bfETYZB/pHDXRcNxNVDsbwuqQL1E22fMkIJHI=";
+  };
+
+  offlineCache = fetchYarnDeps {
+    yarnLock = "${self.src}/yarn.lock";
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/podman-desktop/default.nix:27]: I am 50% sure this triggers IFD

**Comment by SuperSandro2000** [comment]
@ofborg eval

File: pkgs/applications/virtualization/podman-desktop/default.nix (hunk 5 above)
Comment (NickCao) [reply pkgs/applications/virtualization/podman-desktop/default.nix:27]: It's on the negative side.

**Comment by dylanmtaylor** [comment]
Version 0.13.0 is out, new request at #224301

**Pull request merged**